boa_parser/src/parser/statement/labelled_stm/mod.rs

```diff
@@ -65,18 +65,25 @@ where
             // Early Error: It is a Syntax Error if any strict mode source code matches this rule.
             // https://tc39.es/ecma262/#sec-labelled-statements-static-semantics-early-errors
             // https://tc39.es/ecma262/#sec-labelled-function-declarations
-            TokenKind::Keyword((Keyword::Function, _)) if strict => {
+            TokenKind::Keyword((Keyword::Function, _))
+                if cfg!(not(feature = "annex-b")) || strict =>
+            {
                 return Err(Error::general(
-                    "In strict mode code, functions can only be declared at top level or inside a block.",
-                    next_token.span().start()
+                    format!(
+                        "{}functions can only be declared at top level or inside a block.",
+                        if strict { "in strict mode code, " } else { "" },
+                    ),
+                    next_token.span().start(),
```

Comment:
Can this not be `function_declaration_in_if()`?

Reply:
Yep :)